mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import pathlib
+import tempfile
+import zipfile
+from typing import List, Union
+
+import mlflow
+import mlflow.entities
+import mlflow.environment_variables
+import mlflow.types
+
+import mlrun
+from mlrun import MLClientCtx, mlconf
+from mlrun.artifacts import Artifact, ModelArtifact
+from mlrun.features import Feature
+from mlrun.launcher.client import ClientBaseLauncher
+from mlrun.model import RunMetadata, RunObject, RunSpec
+from mlrun.projects import MlrunProject
+from mlrun.track.tracker import Tracker
+from mlrun.utils import now_date
+
+
+class MLFlowTracker(Tracker):
+    """
+    A class for detecting and logging MLFlow runs into MLRun. Notice, only the last active MLFlow run is logged.
+    """
+
+    @classmethod
+    def is_enabled(cls) -> bool:
+        """
+        Checks if tracker is enabled.
+
+        :return: True if the tracking configuration is enabled, False otherwise.
+        """
+        return getattr(mlconf.external_platform_tracking, mlflow.__name__).enabled
+
+    @classmethod
+    def pre_run(cls, context: MLClientCtx):
+        """
+        Initializes the MLFlow tracker, setting the experiment name as configured by the user.
+
+        :param context: Current mlrun context
+        """
+        # Check for a user set experiment name via the environment variable:
+        experiment_name = mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.get()
+
+        # Check if the user configured for matching the experiment name with the context name:
+        if getattr(
+            mlconf.external_platform_tracking, mlflow.__name__
+        ).match_experiment_to_runtime:
+            if experiment_name is not None:
+                context.logger.warn(
+                    f"`mlconf.external_platform_tracking.mlflow.match_experiment_to_runtime` is set to True but the "
+                    f"MLFlow experiment name environment variable ('MLFLOW_EXPERIMENT_NAME') is set for using the "
+                    f"name: '{experiment_name}'. This name will be overriden "
+                    f"with MLRun's runtime name as set in the MLRun configuration: '{context.name}'."
+                )
+            experiment_name = context.name
+            mlflow.set_experiment(experiment_name=experiment_name)
+
+    @classmethod
+    def post_run(cls, context: MLClientCtx):
+        """
+        Performs post-run tasks for logging MLFlow artifacts generated during the run.
+
+        :param context: Current mlrun context
+        """
+        # Get the last run (None means no run was created or found):
```

Comment:
can you tell _when_ or _how_ it might happen? under what circumstances

Reply:
you mean when do we get None? 
an example can br that maybe the mlflow code was badly wrriten and it didn't start the mlflow run

Reply:
or the user has mlconf set to use MLFlow but there was no MLFlow code at all.

Reply:
please do comment that piece of information - helps for debugging